acvm-repo/acvm/src/pwg/blackbox/logic.rs

```diff
@@ -49,11 +51,13 @@ fn solve_logic_opcode<F: AcirField>(
     a: &FunctionInput<F>,
     b: &FunctionInput<F>,
     result: Witness,
+    pedantic_solving: bool,
     logic_op: impl Fn(F, F) -> F,
 ) -> Result<(), OpcodeResolutionError<F>> {
-    // TODO(https://github.com/noir-lang/noir/issues/5985): re-enable these once we figure out how to combine these with existing
+    // TODO(https://github.com/noir-lang/noir/issues/5985): re-enable these by
+    // default once we figure out how to combine these with existing
     // noirc_frontend/noirc_evaluator overflow error messages
-    let skip_bitsize_checks = true;
+    let skip_bitsize_checks = !pedantic_solving;
```

Comment:
I see we have stand-alone functions here, so the `pedantic_solving` can only be a function parameter. Would it make sense to create a `LogicSolver` similar to the `BigintSolver` and make it a config parameter, so as not to complicate the API methods?

Reply:
Or maybe introduce some kind os `SolverContext` that could contain the `initial_witness` and the `pedantic_solving`; most functions I look at take the `&mut WitnessMap` as parameters, and by wrapping these up we could have a bunch of ambient staff we always pass around.

Reply:
I looked into using `WitnessMap`, but 1) I don't want to modify a native type, and 2) `WitnessMap` is used more often than `pedantic_solving`. I already moved `pedantic_solving` to the blackbox/bigint solver structs, so will look into the remaining cases next.

Reply:
To be clear I did not want to suggest modifying `WitnessMap` but to create a `struct SolverContext { initial_witness: WistnessMap, pedantic_solver: bool }` which could be used by the numerous functions that take both of these. If something only takes one we could always pass e.g. `self.ctx.initial_witness`.

Reply:
I'd prefer to keep it as-is since there's only ~5 cases left where `pedantic_solving` is passed along with a `WitnessMap` and I'd have to modify a bunch of places where the `WitnessMap` is created/updated.